src/rpc/blockchain.cpp

```diff
@@ -597,20 +597,32 @@ static RPCHelpMan getblockheader()
     };
 }
 
+void CheckBlockDataAvailability(BlockManager& blockman, const CBlockIndex& blockindex, bool check_for_undo)
+{
+    AssertLockHeld(cs_main);
+    uint32_t flag = check_for_undo ? BLOCK_HAVE_UNDO : BLOCK_HAVE_DATA;
```

Comment:
More or a question: when `check_for_undo` is true doe we actually want to check for both? Having undo and not data set here would seem to be something going weirdly wrong though I guess.

```suggestion
    uint32_t flag = check_for_undo ? BLOCK_HAVE_MASK : BLOCK_HAVE_DATA;
```

Reply:
I agree that it shouldn't be possible to have undo data but no block data.
On the other hand, callers of this function may not even be interested in the block data, and I'm not sure if a RPC call is the right place to check for this inconsistency. At least it's checked in `CheckBlockIndex()` [here](https://github.com/bitcoin/bitcoin/blob/06a9f7789e359131abad2489a221888f711bdc3b/src/validation.cpp#L5326), so I think I'll leave this as is for now.